lib/nonblocking.ml

```diff
@@ -643,6 +643,7 @@ module Make (W : Wait) : S with type 'a future = 'a W.IO.future = struct
       let raw = stmt.Common.Stmt.raw in
       let start = handle_free (B.mysql_stmt_free_result_start raw) in
       let cont s = handle_free (B.mysql_stmt_free_result_cont raw s) in
+      let () = match stmt.Common.Stmt.meta with None -> () | Some { res; _ } -> B.mysql_free_result res in
```

Comment:
This cannot be correct, since `mysql_free_result` is a blocking function, and I think it is only meant for results from non-prepared queries.  Isn't the `mysql_stmt_free_result_start` and `mysql_stmt_free_result_cont` above the correct functions to use?